piksi_tools/console/settings_view.py

```diff
@@ -343,11 +324,12 @@ class SettingsView(HasTraits):
       Show expert settings (defaults to False)
     gui_mode : bool
       ??? (defaults to True)
-    skip : bool
+    skip_read : bool
       Skip reading of the settings (defaults to False). Intended for
-      use when reading from network connections.
+      use when reading from network connections or file.
     """
     show_auto_survey = Bool(False)
+    enable_buttons = Bool(True)
```

Comment:
The current implementation prevents someone from resetting to default settings in the case that the settings could not be discovered.  There  are a lot of cases where settings might not be able to be discovered and we shouldn't neuter any functionality if that happens.

I don't think the "refresh" button or the "reset to default settings" button should be turned off when discovering settings.  I for one use both often when the settings cannot be discovered.

Reply:
I am not sure if I follow. Buttons are disabled only when there's read or write already happening preventing the user for example from spamming the refresh button. Reads and writes will all timeout eventually to make the buttons enabled again.

Reply:
I think that the buttons are disabled while the initial settings discovery is taking place.  Is that not the case?

Reply:
I see, so the variable shall be initialized to `False`.

Reply:
Previous comment was incorrect. Buttons are enabled at start, but disabled during the `read_all` and it takes 2.5s * amount_of_settings to timeout if there's no connection. It is questionable what's the use of enabled buttons if there's no connection.

Reply:
But I'll revert the button disable/enable feature since it's not the point of this PR.